app/Phragile/Burndown.php

```diff
@@ -0,0 +1,107 @@
+<?php
+namespace Phragile;
+
+class Burndown {
+	// TODO: ideally these should come from a cached call to maniphest.querystatuses
+	private static $STATUS_OPEN = ['stalled', 'open'];
+
+	public function __construct(\Sprint $sprint, TaskList $tasks, PhabricatorAPI $phabricator)
+	{
+		$this->sprint = $sprint;
+		$this->tasks = $tasks;
+		$this->phabricator = $phabricator;
+		$this->days = $this->daysInSprint();
+	}
+
+	private function daysInSprint()
+	{
+		$days = [];
+
+		for ($day = strtotime($this->sprint->sprint_start);
+		     $day <= strtotime($this->sprint->sprint_end);
+		     $day += 60*60*24)
+		{
+			$days[] = $day;
+		}
+
+		return $days;
+	}
+
+	public function getDays($format = 'M j')
+	{
+		return array_map(function($day) use($format)
+		{
+			return date($format, $day);
+		}, $this->days);
+	}
+
+	public function closedPerDay()
+	{
+		$closedPerDay = array_fill_keys(array_merge($this->getDays('Y-m-d'), ['before', 'after']), 0);
```

Comment:
I could not figure out why you are merging `['before', 'after']` together with `$this->getDays('Y-m-d')`. The resulting array is rather odd no? So I checked the test to see if it would tell me more, and while it reinforced my impression of primitive obsession, I'm still not entirely sure what this is about.

You can use associative arrays to describe objects, but this is not what arrays are for. Having different types of values in an array without keys is a flag, and having positional arguments in such arrays is an even bigger one.